### DISCUSSION TITLE
Batch Remove Known Background. Is It Possible?

### DISCUSSION THREAD

**Discussion by rustygraham** [ImageMagick/ImageMagick#6944] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10

### Description

I would like to compare 1 image against a folder of images, and 1 at a time delete the difference. 

I found how to do this with a single image:
magick diamond_ore.png stone.png -fuzz 15% -compose ChangeMask -composite diamond_ore_overlay.png

I also know how to modify a folder of images:
magick mogrify -path output -resize 30% -quality 60  input/*.jpg

**Comment by fmw42** [comment, marked as answer]
You likely will need to write a script FOR loop over each image in the folder and do your processing on it.  However, there is no need to invoke mogrify to do the resize. Just include it in your first command  and do that in the script FOR loop.

Alternately, you could do both in a mogrify command. See https://imagemagick.org/Usage/basics/#mogrify_compose for doing an image composite in mogrify.
